corehq/apps/domain/models.py

```diff
@@ -121,6 +121,15 @@ def find_cached(self):
     return find_cached
 
 
+def icds_conditional_session_key():
+    if settings.SERVER_ENVIRONMENT == 'icds':
+        # memoize for process lifecycle
+        return None
+    else:
+        # memoize for request lifecycle
+        return get_session_key
```

Comment:
can you explain a bit about this method?

Reply:
`get_session_key` returns a memoized key that's varied on a UUID that is generated per request cycle. You can refer to its implementation [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/util/quickcache.py#L20) 
```
def get_session_key():
    """
    returns a 7 character string that varies with the current "session" (task or request)
    """
...
```

Reply:
and we are returning in case of ICDS, that results in `domain` returned to be cached for the process lifecycle? which is same as making a global method and memoizing it?

Reply:
Yes, similar, but we want to do this only for ICDS and leave it as it is for other envs. And memoized doesn't allow us to do that conditionally and we still want to cache it for SaaS envs. So quickcache with this conditional cache key allows us to do this.

Reply:
okay thanks
the magic behind the implementation and it not being explicit might make it difficult to get the intend of the change,
how about just making a global method `get_icds_domain_obj` and call that when
domain_name='icds-cas' and `settings. SERVER_ENVIRONMENT in settings.ICDS_ENVS` is true

Reply:
That's something I considered, but that feels more custom than it is now.

Reply:
that would also require changes to every place that uses this

Reply:
ya, i was thinking that would be possible to do.